### ISSUE TITLE
I don't understand use case for array of values. Is it really possible?

### ISSUE THREAD

**Issue by frederikhors** [graphql/dataloader#155]
Sorry to you people, I'm newbie, thanks for your amazing work!

What I don't understand is how to use dataloader when I have **1 to many relationships**.

Here it is a reproduction of my problem: https://enshrined-hydrant.glitch.me.

If you use this query in the Playground:

    query {
      persons {
        name
        bestFriend {
          name
        }
        opponents {
          name
        }
      }
    }

you get values.

But if you open the console log here: https://glitch.com/edit/#!/enshrined-hydrant you can see these database calls I want to avoid:

![image](https://user-images.githubusercontent.com/41120635/45029183-259d4d80-b048-11e8-8eb5-b0a8f4a5809a.png)

My **Person type** is:

    type Person {
      id: ID!
      name: String!
      bestFriend: Person
      opponents: [Person]
    }

I can use dataloader good for `bestFriend: Person` but I don't understand how to use it with `opponents: [Person]`.

As you can see the resolver has to return an *array of values*.

Have you any hint about this?

**Comment by giautm** [comment]
For 1-n relationship, you should get all ids of `Persion` then use [`loadManyKeys`](https://github.com/facebook/dataloader#loadmanykeys) to fetch data.

**Comment by natac13** [comment]
**TL;DR**
If i understand your question correctly, I believe I came across a similar situation. Where I wanted to query for a list of `DBdocuments` based off a single referenced `id`; where I did not have access to a list of ids to pass to `loadmany()`. My solution had me writing a [wrapper package](https://github.com/natac13/Mongoose-Dataloader) for Dataloader to use with Mongoose. 

However to help you out, it is just a matter of how the batch function is set up that you pass to `new Dataloader()`. The difference is minor. For One `id` to one Document I have a batch function as [like this](https://github.com/natac13/Mongoose-Dataloader/blob/master/src/createOneToOneMongooseLoader.js#L35).
And for the One reference 'id' to an [array of Documents](https://github.com/natac13/Mongoose-Dataloader/blob/master/src/createOneToManyMongooseLoader.js#L15)

The situation came up for me while trying to adhere to the MongoDB doc about not embedding 'limitless' list of reference `ids` in a Document. [See Here](https://docs.mongodb.com/manual/tutorial/model-referenced-one-to-many-relationships-between-documents/)

Therefore continuing with the MongoDB Docs example. If the query was:
```
query GetPublisher($id: ID!){
  getPublishers(id: $id) {
    books {
      name
    }
  }
}
```
Set up one to many loader
```
const dataSources = () => ({
  Publisher: PublisherModel.
  BookLoader: new DataLoader(async (keys) => {
    const results = await BookModel.find({ publisher_id: { $in: keys } });
    const hash = R.groupBy(R.prop(field), results);
    // map the keys to meet the Dataloader requirement.
    return keys.map((key) => hash[key.toString()]);
  })
})
```

The resolver would look like:
```
{
  Query: {
    getPublisher: async (root, { id }, { dataSources }) => {
      return dataSources.Publisher.findById(id);
    }
  },
  Publisher: {
    books: async (root, _, { dataSources }) => {
      // returns an array of Book docs that reference the publisher's Id
      const books = await dataSources.BookLoader.load(root.id);
      return books
    }
  }
}
```

For a **Better** explanation I highly recommend this video explaining [Dataloader and the problem it solves](https://www.youtube.com/watch?v=ld2_AS4l19g&list=PLYEivAVs3Cj_XNpv9FliHuyNbnv8AhkaD&index=2&t=150s)

**Comment by motephyr** [comment]
Hi, I made a package to handle has-one, has-many, and belongs-to relationship.
https://github.com/motephyr/dataloader-helper

If there is anything I should pay attention to or improve, please let me know.

**Comment by leebyron** [comment]
Closing this aging issue
